exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -0,0 +1,1365 @@
+			DefaultURL:            kucoinWebsocketURL,
+			RunningURL:            wsRunningEndpoint,
+			Connector:             ku.WsConnect,
+			Subscriber:            ku.Subscribe,
+			Unsubscriber:          ku.Unsubscribe,
+			GenerateSubscriptions: ku.GenerateDefaultSubscriptions,
+			Features:              &ku.Features.Supports.WebsocketCapabilities,
+			OrderbookBufferConfig: buffer.Config{
+				SortBuffer:            true,
+				SortBufferByUpdateIDs: true,
+				UpdateIDProgression:   true,
+			},
+			TradeFeed: ku.Features.Enabled.TradeFeed,
+		})
+	if err != nil {
+		return err
+	}
+	return ku.Websocket.SetupNewConnection(stream.ConnectionSetup{
+		ResponseCheckTimeout: exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:     exch.WebsocketResponseMaxLimit,
+		RateLimit:            500,
+	})
+}
+
+// Start starts the Kucoin go routine
+func (ku *Kucoin) Start(wg *sync.WaitGroup) error {
+	if wg == nil {
+		return fmt.Errorf("%T %w", wg, common.ErrNilPointer)
+	}
+	wg.Add(1)
+	go func() {
+		ku.Run()
+		wg.Done()
+	}()
+	return nil
+}
+
+// Run implements the Kucoin wrapper
+func (ku *Kucoin) Run() {
+	if ku.Verbose {
+		log.Debugf(log.ExchangeSys,
+			"%s Websocket: %s.",
+			ku.Name,
+			common.IsEnabled(ku.Websocket.IsEnabled()))
+		ku.PrintEnabledPairs()
+	}
+
+	if !ku.GetEnabledFeatures().AutoPairUpdates {
+		return
+	}
+
+	err := ku.UpdateTradablePairs(context.TODO(), false)
+	if err != nil {
+		log.Errorf(log.ExchangeSys,
+			"%s failed to update tradable pairs. Err: %s",
+			ku.Name,
+			err)
+	}
+}
+
+// FetchTradablePairs returns a list of the exchanges tradable pairs
+func (ku *Kucoin) FetchTradablePairs(ctx context.Context, assetType asset.Item) (currency.Pairs, error) {
+	switch assetType {
+	case asset.Futures:
+		myPairs, err := ku.GetFuturesOpenContracts(ctx)
+		if err != nil {
+			return nil, err
+		}
+		pairs := make(currency.Pairs, len(myPairs))
+		for x := range myPairs {
+			pairs[x], err = currency.NewPairFromString(strings.ToUpper(myPairs[x].Symbol))
+			if err != nil {
+				return nil, err
+			}
+		}
+		return pairs, nil
+	case asset.Spot, asset.Margin:
+		myPairs, err := ku.GetSymbols(ctx, "")
+		if err != nil {
+			return nil, err
+		}
+		pairs := make(currency.Pairs, len(myPairs))
+		for x := range myPairs {
+			if !myPairs[x].EnableTrading {
+				continue
+			}
+			pairs[x], err = currency.NewPairFromString(strings.ToUpper(myPairs[x].Name))
+			if err != nil {
+				return nil, err
+			}
+		}
+		return pairs, nil
+	default:
+		return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, assetType)
+	}
+}
+
+// UpdateTradablePairs updates the exchanges available pairs and stores
+// them in the exchanges config
+func (ku *Kucoin) UpdateTradablePairs(ctx context.Context, forceUpdate bool) error {
+	assets := ku.GetAssetTypes(false)
+	for a := range assets {
+		pairs, err := ku.FetchTradablePairs(ctx, assets[a])
+		if err != nil {
+			return err
+		}
+		pairs = pairs.RemovePairsByFilter(currency.EMPTYCODE)
```

Comment:
I think you're going to need to leave a comment here. This makes it look like there should be additional filtering within FetchTradablePairs instead of this strange addition here